Move generic mailbox docs from classic to main mailboxes doc (#27493)
akka/akka: akka-docs/src/main/paradox/typed/mailboxes.md

## Selecting what mailbox is used

### Selecting a Mailbox Type for an Actor

> To select mailbox for an actor use `MailboxSelector` to create a `Props` instance for spawning your actor:

Reviewer: Can we add:
```suggestion
To select a mailbox for an actor use `MailboxSelector` to create a `Props` instance for spawning your actor:
```

@@snip [MailboxDocSpec.scala](/akka-actor-typed-tests/src/test/resources/mailbox-config-sample.conf) { }

### Requiring a Message Queue Type for a Dispatcher

A dispatcher may also have a requirement for the mailbox type used by the
actors running on it. An example is the @apidoc[BalancingDispatcher] which requires a
message queue that is thread-safe for multiple concurrent consumers. Such a
requirement is formulated within the dispatcher configuration section like

> this:

Reviewer: I'm not sure the , "like this" is needed

```
my-dispatcher {
  mailbox-requirement = org.example.MyInterface
}
```

The given requirement names a class or interface which will then be ensured to
be a supertype of the message queue’s implementation. In case of a
conflict—e.g. if the actor requires a mailbox type which does not satisfy this
requirement—then actor creation will fail.

### How the Mailbox Type is Selected

When an actor is created, the `ActorRefProvider` first determines the
dispatcher which will execute it. Then the mailbox is determined as follows:

 1. If the actor’s deployment configuration section contains a `mailbox` key
then that names a configuration section describing the mailbox type to be

> used.

Reviewer: The way this is written is confusing to me, can it be clarified - the sentence structure?

 2. If the actor’s `Props` contains a mailbox selection then that names a configuration section describing the
mailbox type to be used (note that this needs to be an absolute config path, 

> for example `myapp.special-mailbox`, and is not nested inside the `akka` namespace).

Reviewer: A bit run-on, perhaps separate into two sentences.

 3. If the dispatcher’s configuration section contains a `mailbox-type` key
the same section will be used to configure the mailbox type.
 4. If the actor requires a mailbox type as described above then the mapping for
that requirement will be used to determine the mailbox type to be used; if
that fails then the dispatcher’s requirement—if any—will be tried instead.
 5. If the dispatcher requires a mailbox type as described above then the
mapping for that requirement will be used to determine the mailbox type to
be used.
 6. The default mailbox `akka.actor.default-mailbox` will be used.

### Default Mailbox

> When the mailbox is not specified as described above the default mailbox

Reviewer: ```suggestion
The default mailbox is used when the mailbox is not specified.
```

> is used. By default it is an unbounded mailbox, which is backed by a

Reviewer: ```suggestion
which is an unbounded mailbox, backed by a
```

`java.util.concurrent.ConcurrentLinkedQueue`.

`SingleConsumerOnlyUnboundedMailbox` is an even more efficient mailbox, and
it can be used as the default mailbox, but it cannot be used with a BalancingDispatcher.

Configuration of `SingleConsumerOnlyUnboundedMailbox` as default mailbox:

```
akka.actor.default-mailbox {
  mailbox-type = "akka.dispatch.SingleConsumerOnlyUnboundedMailbox"
}
```

### Which Configuration is passed to the Mailbox Type

Each mailbox type is implemented by a class which extends `MailboxType`
and takes two constructor arguments: a `ActorSystem.Settings` object and
a `Config` section. The latter is computed by obtaining the named

> configuration section from the actor system’s configuration, overriding its

Reviewer: ```suggestion
configuration section from the `ActorSystem` configuration, overriding its
```

`id` key with the configuration path of the mailbox type and adding a
fall-back to the default mailbox configuration section.

> ## Builtin Mailbox Implementations

Reviewer: ```suggestion
## Mailbox Implementations
```

> Akka comes shipped with a number of mailbox implementations:

Reviewer: ```suggestion
Akka shipps with a number of mailbox implementations:
```

 * 
   **UnboundedMailbox** (default)
    * The default mailbox
    * Backed by a `java.util.concurrent.ConcurrentLinkedQueue`
    * Blocking: No
    * Bounded: No
    * Configuration name: `"unbounded"` or `"akka.dispatch.UnboundedMailbox"`
 * 
   **SingleConsumerOnlyUnboundedMailbox**

>    This queue may or may not be faster than the default one depending on your use-case—be sure to benchmark properly!

Reviewer: ```suggestion
Depending on your use case, this queue may or may not be faster than the default, so you must be sure to benchmark properly.
```

    * Delivery order for messages of equal priority is undefined - contrast with the `BoundedStablePriorityMailbox`
    * Blocking: Yes if used with non-zero `mailbox-push-timeout-time`, otherwise No
    * Bounded: Yes
    * Configuration name: `"akka.dispatch.BoundedPriorityMailbox"`
 * **BoundedStablePriorityMailbox**
    * Backed by a `java.util.PriorityQueue` wrapped in an `akka.util.PriorityQueueStabilizer` and an `akka.util.BoundedBlockingQueue`
    * FIFO order is preserved for messages of equal priority - contrast with the BoundedPriorityMailbox
    * Blocking: Yes if used with non-zero `mailbox-push-timeout-time`, otherwise No
    * Bounded: Yes
    * Configuration name: "akka.dispatch.BoundedStablePriorityMailbox"
 * **BoundedControlAwareMailbox**
    * Delivers messages that extend `akka.dispatch.ControlMessage` with higher priority
    * Backed by two `java.util.concurrent.ConcurrentLinkedQueue` and blocking on enqueue if capacity has been reached
    * Blocking: Yes if used with non-zero `mailbox-push-timeout-time`, otherwise No
    * Bounded: Yes
    * Configuration name: "akka.dispatch.BoundedControlAwareMailbox"

> ## Creating your own Mailbox type

Reviewer: ```suggestion
## Custom Mailbox type
```

> An example is worth a thousand quacks:

Reviewer: ```suggestion
The best way to show how to roll your own Mailbox type is by example:
```
